concent_api/middleman/queue_operations.py

```diff
@@ -162,6 +169,30 @@
         response_queue.task_done()
 
 
+async def is_authenticated(reader: StreamReader, writer: StreamWriter) -> bool:
+    challenge = create_random_challenge()
+
+    request_id = randint(1, 7777)
+    frame = AuthenticationChallengeFrame(challenge, request_id)
+    await send_over_stream_async(frame, writer, settings.CONCENT_PRIVATE_KEY)
+    logger.info(f"Challenge has been sent for request ID: {request_id}.")
+    received_frame = await handle_frame_receive_async(reader, settings.SIGNING_SERVICE_PUBLIC_KEY)
+    logger.info(f"Response has been received for request ID: {request_id}.")
+    if received_frame.request_id != request_id:
```

Comment:
Below you are check if `isinstance(received_frame, AuthenticationResponseFrame)`. Maybe it would be good to check it first and as second check  `if received_frame.request_id != request_id:`. If received_frame won't be `AuthenticationResponseFrame` instance it may don't have field `request_id` (received_frame.request_id ) and it may be cause of error.

Reply:
I also think that two checks should be combined together if they result in the same (here: `return False`).

Reply:
done